lib/graph3d/DataGroup.js

```diff
@@ -0,0 +1,240 @@
+var DataSet  = require('../DataSet');
+var DataView = require('../DataView');
+var Point3d  = require('./Point3d');
+var Range    = require('./Range');
+
+
+function DataGroup() {
+  this.dataTable = null;  // The original data table
+}
+
+
+/**
+ * Initialize the data from the data table. Calculate minimum and maximum values
+ * and column index values
+ * @param {Array | DataSet | DataView} rawData The data containing the items for
+ *                                             the Graph.
+ * @param {Number}                     style   Style Number
+ */
+DataGroup.prototype.dataInitialize = function(graph3d, rawData, style) {
+  var me = this;
+
+  // unsubscribe from the dataTable
+  if (this.dataSet) {
+    this.dataSet.off('*', this._onChange);
+  }
+
+  if (rawData === undefined)
+    return;
+
+  if (Array.isArray(rawData)) {
+    rawData = new DataSet(rawData);
+  }
+
+  var data;
+  if (rawData instanceof DataSet || rawData instanceof DataView) {
+    data = rawData.get();
+  }
+  else {
+    throw new Error('Array, DataSet, or DataView expected');
+  }
+
+  if (data.length == 0)
+    return;
+
+  this.dataSet = rawData;
+  this.dataTable = data;
+
+  // subscribe to changes in the dataset
+  this._onChange = function () {
+    me.setData(me.dataSet);
+  };
+  this.dataSet.on('*', this._onChange);
+
+  // determine the location of x,y,z,value,filter columns
+  this.colX = 'x';
+  this.colY = 'y';
+  this.colZ = 'z';
+
+
+  var withBars = graph3d.hasBars(style);
+
+  // determine barWidth from data
+  if (withBars) {
+    if (graph3d.defaultXBarWidth !== undefined) {
+      this.xBarWidth = graph3d.defaultXBarWidth;
+    }
+    else {
+      this.xBarWidth = this.getSmallestDifference(data, this.colX) || 1;
+    }
+
+    if (graph3d.defaultYBarWidth !== undefined) {
+      this.yBarWidth = graph3d.defaultYBarWidth;
+    }
+    else {
+      this.yBarWidth = this.getSmallestDifference(data, this.colY) || 1;
+    }
+  }
+
+  // calculate minimums and maximums
+  var NUMSTEPS = 5;
+
+  var xRange = this.getColumnRange(data, this.colX);
```

Comment:
Is there any way we can shorten the following lines? Looks quite similar to me. Only the axis is different.